### ISSUE TITLE
No package.json (or package.yaml, or package.json5) was found in "/Users/yutengjing/Library/pnpm/global/5"

### ISSUE THREAD

**Issue by tjx666** [pnpm/pnpm#4645]
### Expected behavior:

No error, and list the packages installed in global

### Actual behavior:

<img width="1033" alt="image" src="https://user-images.githubusercontent.com/41773861/165993645-cfef26a5-3e7c-4bbd-a537-c75514855b1f.png">

The error message said there is no `/Users/yutengjing/Library/pnpm/global/5`, is the path wrong?
I see that there is another folder:

<img width="1048" alt="image" src="https://user-images.githubusercontent.com/41773861/165993944-cfd1020a-6649-4b66-9118-f6e0a0119782.png">

### Additional information:

- MacOS: 12.3.1
- Node: v16.15.0

**Comment by zkochan** [comment]
I renamed the folder in one of the rc versions.

**Comment by eran-or** [comment]
npx pnpm install --global pnpm
Fixed for me the problem

**Comment by mcmxcdev** [comment]
Seems like this issue was only relevant in one of the pnpm v7 rc versions.

**Comment by amiiit** [comment]
Happened to me just now, first time ever I used pnpm on a mac, I way trying to run:

```
pnpm create next-app
```

The suggestion above fixed the issue for me:

```
npx pnpm install --global pnpm
```

**Comment by FabulousSoldier** [comment]
Had the same problem in Ubuntu 22.04.1 LTS with pnpm 7.13.6.

`npx pnpm install --global pnpm`

fixed it for me as well.

**Comment by thejohnhoffer** [comment]
I saw this issue after globally linking a package with `pnpm link -g .` instead of `pnpm link -g`. After running `npx pnpm install --global pnpm` to fix the issue, then correctly running `pnpm link -g`, I did not see this issue.

**Comment by nx-dennytrebbin** [comment]
I'm using `asdf` to manage `pnpm` versions and see this error now when trying to update `pnpm`.
Installing something globally is not a good recommendation unless you work in just one project at any time.

**Comment by kiwi44** [comment]
For me it was a case of moving the project files up one folder level.

when I received the error:
**Command: `pnpm dev                              
**Error:** ERR_PNPM_NO_IMPORTER_MANIFEST_FOUND  No package.json (or package.yaml, or package.json5) was found in [my directory name]` the above solutions were not useful. 

The solution for me was simpler, to move the project files (the files created via the Next start project commands) up one level into the root folder. Then, running

'pnpm dev' worked thereafter.

HTH

**Comment by Amit0617** [comment]
> The solution for me was simpler, to move the project files (the files created via the Next start project commands) up one level into the root folder...

Lol, basically if you had run `pnpm dev` **inside** the directory containing project files. It would have worked. 
This thread was for issue in installation of **global** packages that is faced after fresh installation of `pnpm`.

**Comment by gideonpraise** [comment]
Just to add to this, I have found that most people that have this issue now seems to be running the wrong command.

Example:
Running `pnpm create-next-app@latest` or `pnpm create-next-app` will cause that error.

The correct command is `pnpm create next-app`.

The pnpm package manager exports a `create` command in its cli when initializing project from a starter kits or template.
Can read here [https://pnpm.io/cli/create](https://pnpm.io/cli/create)

Hope that helps someone!

**Comment by Amit0617** [comment]
> Just to add to this, I have found that most people that have this issue now seems to be running the wrong command.
> 
> Example: Running `pnpm create-next-app@latest` or `pnpm create-next-app` will cause that error.
> 
> The correct command is `pnpm create next-app`.
> 
> The pnpm package manager exports a `create` command in its cli when initializing project from a starter kits or template. Can read here https://pnpm.io/cli/create
> 
> Hope that helps someone!

You just solved your own problem which was actually ignorant behaviour of yours towards its usage documentation and then posting it under an error which has literally no connection to your “suggestions” and hoping it helps someone. Great!

**Comment by justinphilpott** [comment]
@Amit0617 Sarcasm isn't really that friendly is it.

**Comment by Amit0617** [comment]
@justinphilpott yeah... I was simply criticising a wrong stuff, but then I felt it was harsh. So, tried to add a bit of sarcasm, which apparently became harsh sarcasm.
Looks like that comment is already deleted by author. At least, the quoted reply gives enough context. I am a hardcore Linux guy, who feels it's your responsibility to search for your answers, read docs and wikis before screaming on internet.
You won't believe how much bad I am feeling commenting this irrelevant stuff on the comments of this issue. If there was personal message kind of stuff on GitHub, I would have sent you this in private.

**Comment by acemtaylor** [comment]
In saying that I just found a possible solution on stack overflow

_think, `npm init` will create your missing package.json file. It works for me for the same case_ [src](https://stackoverflow.com/questions/9484829/npm-cant-find-package-json)

**Comment by specious** [comment]
I've been consistently running into this on WSL (Ubuntu and Debian), trying to install and run global packages out of the box.

I install pnpm using the official installer:

```
curl -fsSL https://get.pnpm.io/install.sh | sh -
```

I let it add the lines to my shell configuration:

```bash
export PNPM_HOME="/home/me/.local/share/pnpm"
case ":$PATH:" in
  *":$PNPM_HOME:"*) ;;
  *) export PATH="$PNPM_HOME:$PATH" ;;
esac
```

`pnpm` works and I run e.g. `pnpm i -g qrcode-terminal`:

```
┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( pnpm --version
10.33.0

┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( pnpm i -g qrcode-terminal
Packages: +1
+
Progress: resolved 1, reused 0, downloaded 1, added 1, done

/home/me/.local/share/pnpm/global/5:
+ qrcode-terminal 0.12.0

Done in 698ms using pnpm v10.33.0
```

The executable is in the path, but it's a pnpm shim that acts as a proxy for the actual executable and it doesn't behave correctly:

```
┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( which qrcode-terminal
/home/me/.local/share/pnpm/qrcode-terminal

┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( qrcode-terminal
 ERR_PNPM_NO_IMPORTER_MANIFEST_FOUND  No package.json (or package.yaml, or package.json5) was found in "/home/me".
```

I can run it directly:

```
┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( $PNPM_HOME/global/5/node_modules/qrcode-terminal/bin/qrcode-terminal.js

Usage: qrcode-terminal <message>

Options:
  -h, --help           output usage information
  -v, --version        output version number

Examples:

  $ qrcode-terminal hello
  $ qrcode-terminal "hello world"
```

I found two solutions to the problem:

```
┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( pnpm un -g qrcode-terminal
 WARN  1 deprecated subdependencies found: node-domexception@1.0.0
Packages: -1
-
Progress: resolved 15, reused 15, downloaded 0, added 0, done

/home/me/.local/share/pnpm/global/5:
- qrcode-terminal 0.12.0

Done in 1s using pnpm v10.33.0

┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( bun i -g qrcode-terminal
bun add v1.3.11 (af24e281)

installed qrcode-terminal@0.12.0 with binaries:
 - qrcode-terminal

1 package installed [205.00ms]

┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( qrcode-terminal

Usage: qrcode-terminal <message>

Options:
  -h, --help           output usage information
  -v, --version        output version number

Examples:

  $ qrcode-terminal hello
  $ qrcode-terminal "hello world"
```

(which executes much faster)

But also:

```
┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( pnpm i -g pnpm@latest
Packages: +1
+
Progress: resolved 1, reused 0, downloaded 1, added 1, done

/home/me/.local/share/pnpm/global/5:
+ pnpm 10.33.0

Done in 1.5s using pnpm v10.33.0

┌───=[ me :: aspire ]-( 0 )-[ ~ ]
└──( pnpm --version
10.33.0
```

The `pnpm` version remained the same, but global packages run all of a sudden.

My current recommendation is `bun` for globally installed executables and `pnpm` as the go-to for local project dependencies.

**Comment by zkochan** [comment]
That last issue is fixed in pnpm 10.33.2. But it was not related to the original issue.
